docs/application/native/guides/ui/tizen-ws-shell.md

```diff
@@ -0,0 +1,151 @@
+
+
+## Prerequisites
+
+To use the functions and structures of the tzsh-quickpanel API (in [mobile](../../api/mobile/latest/group__TIZEN__WS__SHELL__QUICKPANEL__MODULE.html) and [wearable](../../api/wearable/latest/group__TIZEN__WS__SHELL__QUICKPANEL__MODULE.html) applications), include the `<tzsh_quickpanel.h>` header file in your application:
+
+```
+#include <tzsh_quickpanel.h>
+```
+
+
+## Creating TZSH Quickpanel Handler
+Once you have created the main window for your application, call the `tzsh_quickpanel_create_with_type` function with a native window ID to create `tzsh_quickpanel_h` structure (in [mobile](../../api/mobile/latest/group__TIZEN__WS__SHELL__QUICKPANEL__MODULE.html#gaaa00e8e25b43c9538ca188bc43bdb3ac) and [wearable](../../api/wearable/latest/group__TIZEN__WS__SHELL__QUICKPANEL__MODULE.html#gaaa00e8e25b43c9538ca188bc43bdb3ac) applications).
+
+```
+#include <Elementary.h>
+
+static void
+init(const char *name)
+{
+   Evas_Object *main_win;
+
+   /* create main window for the application */
+   main_win = elm_win_util_standard_add(name, name);
+
+   /* set up main window */
+   ...
+   evas_object_show(main_win);
+
+
+   tzsh_h tzsh;
+   tzsh_quickpanel_h tzsh_qp;
+   tzsh_window tz_win;
+
+   /* Get native window ID of main window */
+   tz_win = elm_win_window_id_get(main_win);
+
+   /* Create tzsh_h structure */
+   tzsh = tzsh_create(TZSH_TOOLKIT_TYPE_EFL);
+
+   /* Create tzsh_quickpanel_h structure */
+   tzsh_qp = tzsh_quickpanel_create_with_type(tzsh, tz_win, TZSH_QUICKPANEL_TYPE_SYSTEM_DEFAULT);
+```
+
+
+## Showing and Hiding Quickpanel Window
+
+To show and hide quickpanel window when your application's window is activated, call the following functions:
+
+```
+   ...
+
+   if (show)
+     tzsh_quickpanel_show(tzsh_qp); /* To show quickpanel window */
+   else
+     tzsh_quickpanel_hide(tzsh_qp); /* To hide quickpanel window */
+
+   ...
+```
+
+## Getting Visible Status of Quickpanel Window
+To know the state of current visibility of qp service's window, call the `tzsh_quickpanel_visible_get` function. The quickpanel service's window may be visible or not visible on the screen.
+
+```
+   ...
+   tzsh_quickpanel_state_visible_e state;
+
+   tzsh_quickpanel_visible_get(tzsh_qp, &state);
+   if (state == TZSH_QUICKPANEL_STATE_VISIBLE_SHOWN)
+     /* visible state */
+     ;
+   else if (state == TZSH_QUICKPANEL_STATE_VISIBLE_HIDDEN)
+     /* invisible state */
+     ;
+   else
+     /* error */
+     ;
+```
+
+
+## Registering Changed Event for the Quickpanel Window
+To be notified of state changes, implement the appropriate event callback function and call the `tzsh_quickpanel_event_handler_add` function with that. If you want to change your application's behavior to match visibility of the quicknapel service's window, then you should handle state change event as following:
```

Comment:
To be notified about the state changes, implement the appropriate event callback function and call the `tzsh_quickpanel_event_handler_add` function with that. If you want to change your application's behavior to match visibility of the quick-panel service window, you need to handle state change event as shown in the following code snippet:

Reply:
How about simplifying the sentence, as like "you need to handle state change event as below:"?

Reply:
I like @mijin85cho's suggestion. How about your opinion @nawab-samsung?

Reply:
@mijin85cho yes that's fine. 
@gl-lee , Kindly update: 
To be notified about the state changes, implement the appropriate event callback function and call the `tzsh_quickpanel_event_handler_add` function with that. If you want to change your application's behaviour to match visibility of the quickpanel service window, you need to handle state change event as follows:

Reply:
Fine